test/versioned/mongodb-esm/common.cjs

```diff
@@ -13,7 +13,7 @@ const urltils = require('../../../lib/util/urltils')
 
 const MONGO_SEGMENT_RE = /^Datastore\/.*?\/MongoDB/
 const TRANSACTION_NAME = 'mongo test'
-const DB_NAME = 'integration'
+const DB_NAME = 'esmIntegration'
```

Comment:
this is suspect is the actual fix. for esm tests connect to a diff database so when in db.tap.js we drop the database it won't cause random issues for the other test suite